### ISSUE TITLE
kaltura-dwh-config.sh doesn't check properly if kalturadw database exists

### ISSUE THREAD

**Issue by wzur** [kaltura/platform-install-packages#494]
Hi there,

`kaltura-dwh-config.sh` checks if _kalturadw_ has been populated, before it runs `dwh_setup.sh`

Unfortunately, when datawarehouse MySQL is on a different server than the main Kaltura database, it leads to an error.

_mysql_ command returns non-zero exit code, as 'kalturadw' database doesn't exist yet, which coupled with '-e' switch for bash makes the configuration script to return at that point.

Instead _mysql_ should first check if that database exists, and only then checks if that database has any tables inside.

As a side note - I didn't see any guidelines for that setup - DWH database on a separate host than DB1_HOST, in the documentation. Any pointers would be greatly appreciated.

Waldemar

**Comment by jessp01** [comment]
Hello,

We've never actually tested a configuration where the DWH DB resides on a separate host than the operational 'kaltura' one.
I agree that such a use case should generally be supported, however.

The first issue I see, which currently is the only one I can think of, is that in /opt/kaltura/bin/kaltura-db-config.sh, all DBs are created on $DB1_HOST which is the operational DB, whereas, in the scenario you're describing, kaltura_sphinx_log kalturadw kalturadw_ds kalturadw_bisources kalturalog should actually be created on $DWH_HOST instead, which is an easy enough change to /opt/kaltura/bin/db_actions.rc, separating:
DBS="kaltura kaltura_sphinx_log kalturadw kalturadw_ds kalturadw_bisources kalturalog"
into two separate vars, i,e:
OPERATIONAL_DBS="kaltura"
DWH_DBS="kaltura_sphinx_log kalturadw kalturadw_ds kalturadw_bisources kalturalog"
and then change /opt/kaltura/bin/kaltura-db-config.sh accordingly.

If you'd be willing to test these changes and confirm all works well, I'll be happy to review a pull request from you and merge it.

Thanks,

**Comment by wzur** [comment]
Well - it didn't work. I went ahead and created a cluster with a separate database server for DWH, and did a manual step to create all databases, regardless of the installation scripts. However KMC didn't work correctly in that scenario, anyway.

So perhaps it would be better to get rid of that variable altogether?

W.

**Comment by KalturaCommunity** [comment]
Can you please explain what did not work in KMC? I presume the issue was 
somewhere in alpha/apps/kaltura/lib/reports/myReportsMgr.class.php and in 
/opt/kaltura/log/kaltura_api_v3.log something like 'Can\'t use foo : ' was 
outputted...

That code takes the DSN from /opt/kaltura/app/configurations/local.ini:
[reports_db_config]
host = 
user = etl
port = 3306
password = 
db_name = kalturadw

I'd appreciate it if you can check that the connection string is correct 
and let me know as to the exact error so we can resolve it. Like I said 
before, nobody really expressed much interest in this configuration before 
since for most deployments, running all DBs on the same MySQL host makes 
sense but still, always better to support than not to support:)


May the source be with you,

Jess Portnoy

Kaltura Connect Virtual Summits 2015 - now on VOD! [connect.mediaspace.kaltura.com]

**Comment by wzur** [comment]
Hi Jess,

KMC just showed 'internal server error', but I didn't investigate at the backend why it failed.
I dropped the idea of having two MySQL databases when I realised DWH_HOST is used only to point to the MySQL database.

Personally - I think DWH_HOST should be dropped, just like SPINHX_DB_HOST.

I'll create a cluster again, however and I'll provide you with the required information a bit later.

Waldemar
